lib/docgen/class_template.md.erb

```diff
@@ -0,0 +1,81 @@
+<%= kind %> <%= title %>
+========================
+<%= comment %>
+<% if constants.any? -%>
+- [Constants](#constants)
+<%- end -%>
+<%- if attributes.any? -%>
+- [Attributes](#attributes)
+<%- end -%>
+<%- if class_methods.any? -%>
+- [Class Methods](#class-methods)
+<%- class_methods.each do |method| -%>
+  - [<%= method.title %>](#<%= method.title.gsub("?", "") %>)
+<%- end -%>
+<%- end -%>
+<%- if instance_methods.any? -%>
+- [Instance Methods](#instance-methods)
+<%- instance_methods.each do |method| -%>
+  - [<%= method.title %>](#<%= method.title.gsub("?", "") %>)
+<%- end -%>
+<%- end -%>
+<%- if included.any? -%>
+- [Includes](#includes)
+<%- end -%>
+<%- if extended.any? -%>
+- [Extends](#extends)
+<%- end -%>
+<%- if constants.any? %>
+# Constants
+<%- constants.each do |const| -%>
+- `<%= const.title %>` <%= const.comment %>
+<%- end -%>
+<%- end -%>
+<%- if attributes.any? %>
+# Attributes
+<%- attributes.each do |attr| -%>
+- `<%= attr.title %>` <%= attr.comment %>
+<%- end -%>
+<%- end -%>
+<%- if class_methods.any? %>
+# Class Methods
+<% class_methods.each do |method| %>
+---
```

Comment:
If you move the `---` to the end of each iteration instead of the beginning, then we can avoid the double underline:

![image](https://user-images.githubusercontent.com/71433/79920680-a5835180-83fe-11ea-915b-e85bb01a41be.png)